libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/installer/ComponentInstaller.kt

```diff
@@ -0,0 +1,73 @@
+package com.mapbox.navigation.ui.maps.installer
+
+import android.content.Context
+import com.mapbox.maps.MapView
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.ui.base.installer.ComponentInstaller
+import com.mapbox.navigation.ui.base.installer.Installation
+import com.mapbox.navigation.ui.maps.internal.ui.RouteArrowComponent
+import com.mapbox.navigation.ui.maps.internal.ui.RouteLineComponent
+import com.mapbox.navigation.ui.maps.route.arrow.api.MapboxRouteArrowView
+import com.mapbox.navigation.ui.maps.route.arrow.model.RouteArrowOptions
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
+import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineResources
+
+/**
+ * Install component that renders route line on the map.
+ *
+ * The installed component:
+ * - renders route lines returned by [RoutesObserver]
+ * - vanishes the traveled portion of the route line (if enabled via [MapboxRouteLineOptions] in the configuration)
+ * - selects alternative route on map click
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+fun ComponentInstaller.routeLine(
+    mapView: MapView,
+    config: RouteLineComponentConfig.() -> Unit = {}
+): Installation {
+    val componentConfig = RouteLineComponentConfig(mapView.context).apply(config)
+    return component(RouteLineComponent(mapView, componentConfig.options))
```

Comment:
Do we need to register it to `MapboxNavigationApp`? Same for `routeArrow` below

@tomaszrybakiewicz

Reply:
Nope. `NavigationComponents` will do it for us.